reference/concepts.md

```diff
@@ -0,0 +1,105 @@
+# Haskell reference
+
+## Concepts
+
+The Haskell concept exercises are based on concepts. The list below contains the concepts that have been identified for the Haskell language.
+
+### Language-unique
+
+- ~~Active patterns~~ Pattern synonyms? That seems a stretch.
+- ~~Computation expressions~~ do-notation
+- Dependency order
+
+### Functional
+
+- Expression-oriented (no statements at all)
+- Functions
+  - Anonymous functions
+  - Currying
+  - Function composition
+  - Higher-order functions
+  - ~~Implicit returns~~ (no returns at all)
+  - Nested functions 🤔
```

Comment:
Ah, sounds like simply **local definitions** + **functions are values** then, [in-Haskell-universe](https://en.wiktionary.org/wiki/in-universe). I would be hesitant to tell new programmers that a [class distinction](https://en.wikipedia.org/wiki/First-class_citizen) _could_ exists but in fact doesn't, because it might lead to unnecessary confusion.

Reply:
Yeah I think those are the same thing when combined. Feel free to remove concepts that you feel don't apply to Haskell.